code-challenges.test.js

```diff
@@ -10,41 +12,190 @@
+        // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+        const output = ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
+        expect(twoToOne(dataArray1,dataArray2)).toEqual(output)
+    })
+})
+// Errors received:
+// ReferenceError: Cannot access 'filterDuplicates' before initialization
 
 // b) Create the function that makes the test pass.
+// A function that takes 2 arrays
+const twoToOne = (arr1,arr2) => {
+
+    // A function that takes in an array and returns it without duplicate entries
+    const noDuplicates = (array1) => {
+
+        // Iterate over the array in ascending order
+        for (let i = 0 ; i < array1.length ; i++){
+
+            // Iterate over the array in descending order
+            for (let j = array1.length-1 ; j > -1  ; j--){
+
+                // Check if a value of a certain index has a duplicate not itself
+                if (array1[i] == array1[j] && i != j){
+
+                    // If a duplicate exists, splice it away
+                    array1.splice(j,1)
+                }
+            }
+        }
+        // Return the argument array without duplicates
+        return array1
+    }
+
+    // Ensure no duplicates in 1st argument array
+    // const arr3 = noDuplicates(arr1)
+
+    // Ensure no duplicates in 2nd argument array
+    // const arr4 = noDuplicates(arr2)
+
+    // Ensure no duplicates in concatenated no-duplicate argument arrays
+    // const arr5 = arr3.concat(arr4)
+
+    // Ensure no duplicates in concatenated no-duplicate arrays and return
+    // return noDuplicates(arr5)
+
+    // Below is the above 4 lines of code in one
+    return noDuplicates((noDuplicates(arr1)).concat(noDuplicates(arr2)))
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+
+// Errors generated:
+// TypeError: args.flat is not iterable
+
+// STRETCH function below, accepts variable amount of arguments
+const manyToOne = (...args) => {
+
+    // A function that takes in an array and returns it without duplicate entries
+    const noDuplicates = (array1) => {
+
+        // Iterate over the array in ascending order
+        for (let i = 0 ; i < array1.length ; i++){
+
+            // Iterate over the array in descending order
+            for (let j = array1.length-1 ; j > -1  ; j--){
+
+                // Check if a value of a certain index has a duplicate not itself
+                if (array1[i] == array1[j] && i != j){
+
+                    // If a duplicate exists, splice it away
+                    array1.splice(j,1)
+                }
+            }
+        }
+        // Return the argument array without duplicates
+        return array1
+    }
+    // Use .flat() method to concatenate multiple arrays as arguments
+    // return the flattened arrays without duplicates
+    return noDuplicates([...args.flat()])
+}
+
+// Describe block for STRETCH implementation of variable argument quantity
```

Comment:
WOW!  You really went all out with this problem and completely crushed it!!!  Excellent work on all parts - from the tests, pseudocode and to the code, this is amazing! 👏